module/VuFind/src/VuFind/RecordTab/HoldingsILS.php

```diff
@@ -100,13 +100,31 @@ public function getDescription()
     public function getUniqueCallNumbers($items)
     {
         $callNos = [];
-        foreach ($items as $item) {
+        foreach ($items as $i => $item) {
             if (isset($item['callnumber']) && strlen($item['callnumber']) > 0) {
-                $callNos[] = $item['callnumber'];
+                $prefix = $item['callnumber_prefix'] ?? '';
+                $callnumber = $item['callnumber'];
+                $callNos[$i]['callnumber'] = $callnumber;
```

Comment:
I think indexing directly into `[$i]` without first initializing an array may cause some warnings in some of the static analysis tools we are introducing to VuFind. It's usually better to initialize things explicitly. I wonder, in this case, if it would actually be more readable to do the work in variables and then use `compact` to turn them into an array... e.g.:

<pre>
$prefix = $item['callnumber_prefix'] ?? '';
$callnumber = $item['callnumber'];
$display = $prefix ? $prefix . ' ' . $callnumber : $callnumber;
$callNos[] = compact('callnumber', 'display', 'prefix');
</pre>

(This also eliminates some redundant calculations around prefix).